fbgemm_gpu/codegen/embedding_backward_split_indice_weights_template.cu

```diff
@@ -253,7 +265,7 @@ Tensor {{ ddesc }}_embedding_codegen_grad_indice_weights{{ vdesc }}_cuda(
     const auto total_B = offsets.size(0) - 1;
     TORCH_CHECK_GE(total_B, 0);
     TORCH_CHECK_LE(max_D, {{ max_embedding_dim }});
-    auto grad_indice_weights = empty_like(indices, indices.options().dtype(at::toAccumulateType(grad_output.scalar_type(), true)));
+    auto grad_indice_weights = empty_like(indices, indices.options().dtype(at::toAccumulateType(aligned_grad_output.scalar_type(), true)));
```

Comment:
I'm not going to carefully audit that you updated all the downstream use sites. To make it obvious you didn't do it wrong, change the input name and then once you align grad output, assign it to `grad_output`, no diff afterwards.